tools/pylib/boututils/datafile.py

```diff
@@ -319,9 +319,13 @@ def __getitem__(self, name):
     def __setitem__(self, key, value):
         self.impl.__setitem__(key, value)
 
-    def attributes(self, varname):
+    def attributes(self, varname=""):
         """Return a dictionary of attributes
 
+        These are attributes for varname if varname is specified. If varname is
+        the empty string (default case) then the global attributes of the file
+        are returned.
```

Comment:
Does this actually work? I can't see anything that handles the empty string in the NetCDF/HDF5 implementations of `attributes`. It looks like they both try and find attributes for a variable named `""`, i.e. the empty string. As there likely *isn't* a variable called this, they return `None`. I might be wrong, I don't have a file with file-level attributes to hand!

Also, I'm assuming "global attributes" means "file attributes"?